src/SharpLearning.Optimization/GlobalizedBoundedNelderMeadOptimizer.cs

```diff
@@ -109,7 +113,7 @@ public OptimizerResult[] Optimize(Func<double[], OptimizerResult> functionToMini
                 var iterationsWithoutImprovement = 0;
                 var results = new List<OptimizerResult> { new OptimizerResult(prevBest.ParameterSet, prevBest.Error) };
```

Comment:
@jameschch The result list should probably be changed to a concurrent collection to avoid potential data curruptions, if multiple threads are adding to it.

Reply:
Yes, I did add a manual lock to the other optimizer for this purpose but missing it here.